apps/backoffice-v2/src/lib/blocks/hooks/useDirectorsBlocks/useDirectorsBlocks.tsx

```diff
@@ -148,33 +244,45 @@ export const useDirectorsBlocks = ({
                 })
                 .build()
                 .flat(1);
+
+              return approvedBadgeBlock;
+            } else {
```

Comment:
I think its important for these functions to work like a switch case with a default over using `else`. Explicitly define a cell/block for each `decisionStatus` and return nothing, some fallback, or throw an error if no condition matched.

Reply:
Switch case won't work here because some conditions having multiple levels of check. `default` not needed as well, its either return control/badge or return nothing.

Reply:
Will be refactored once more statuses are introduced.